src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,219 @@
+        try {
+            String[] items = input.substring(6).split(" /at ");
+            insertTask(new Event(items[0], false, items[1]));
+        } catch (StringIndexOutOfBoundsException | ArrayIndexOutOfBoundsException e) {
+            System.out.println("description cannot be empty");
+        } catch (DukeException e) {
+            System.out.println("please enter a valid date format.");
+            System.out.println("date and time: dd/mm/YYYY hh:mm");
+        }
+    }
+
+    /**
+     * Inserts task into given task list.
+     *
+     * @param task Task to be inserted into the list.
+     */
+    protected void insertTask(Task task) {
+        tasks.add(task);
+        System.out.println("added: ");
+        System.out.println("\t" + task);
+        System.out.format("you have %d task(s) in the list\n", tasks.size());
+    }
+
+    /**
+     * Parses String in the form of mark x, marking index x in the list as done.
+     *
+     * @param input String that contains information about item to be marked done.
+     */
+    protected void markItemDone(String input) {
+        try {
+            String[] words = input.split(" ");
+            if (words.length > 2) {
+                throw new DukeException("too many parameters!");
+            }
+            int index = Integer.parseInt(words[1]);
+            tasks.get(index - 1).markDone();
+            System.out.println("cool, this task is marked as done");
+            System.out.println("\t" + tasks.get(index - 1));
+        } catch (NumberFormatException | ArrayIndexOutOfBoundsException | DukeException e) {
+            System.out.println("format: mark <number>");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("enter a valid index");
+        }
+    }
+
+    /**
+     * Parses String in the form of mark x, marking index x in the list as not done.
+     *
+     * @param input String that contains information about item to be marked not done.
+     */
+    protected void markItemUndone(String input) {
+        try {
+            String[] words = input.split(" ");
+            if (words.length > 2) {
+                throw new DukeException("too many parameters!");
+            }
+            int index = Integer.parseInt(words[1]);
+            tasks.get(index - 1).markUndone();
+            System.out.println("ok, this task is marked as not done yet");
+            System.out.println("\t" + tasks.get(index - 1));
+        } catch (NumberFormatException | ArrayIndexOutOfBoundsException | DukeException e) {
+            System.out.println("format: mark <number>");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("enter a valid index");
+        }
+    }
+
+    /**
+     * Parses String in the form of mark x, deleting task at index x in the list.
+     *
+     * @param input String that contains information about item to be deleted.
+     */
+    protected void deleteTask(String input) {
```

Comment:
So many methods .... Maybe you could use a new class to handle these methods : ) it will be more readable : )